reader.go

```diff
@@ -400,6 +400,11 @@ type ReaderConfig struct {
 	// Default: 10s
 	MaxWait time.Duration
 
+	// MaxSafetyTimeout amount of time to wait to fetch message from kafka messages batch.
+	//
+	// Default: 10s
+	MaxSafetyTimeout time.Duration
```

Comment:
What do you think about renaming this to `ReadBatchTimeout` since it's controlling the time we allocate to read batches off of the network connection?

Reply:
I like it it's more descriptive, happy to change